libcontainer/container_linux.go

```diff
@@ -960,12 +974,15 @@ func (c *linuxContainer) criuSwrk(process *Process, req *criurpc.CriuReq, opts *
 
 	// cmd.Wait() waits cmd.goroutines which are used for proxying file descriptors.
 	// Here we want to wait only the CRIU process.
-	st, err := cmd.Process.Wait()
-	if err != nil {
-		return err
-	}
-	if !st.Success() {
-		return fmt.Errorf("criu failed: %s\nlog file: %s", st.String(), logPath)
+	reqType := req.GetType()
+	if reqType != criurpc.CriuReqType_PRE_DUMP {
+		st, err := cmd.Process.Wait()
```

Comment:
`cat /proc/[pid]/stack` might help you out. Note that doing an `strace` of a multithreaded process is not recommended (when you do `execve` there are some Linux kernel bugs which mean that you sometimes lock up the process because the other threads in the process become zombies and the `execve` is blocked on them dying).